src/main/java/org/mskcc/cmo/messaging/impl/JSGatewayImpl.java

```diff
@@ -154,15 +157,12 @@ public void subscribe(String subject, Class messageClass,
             Dispatcher dispatcher = natsConnection.createDispatcher();
             ConsumerConfiguration consumerConfig = ConsumerConfiguration.builder()
                     .durable(consumerName)
-                    .filterSubject(filterSubject)
-                    .ackPolicy(AckPolicy.All)
-                    .deliverPolicy(DeliverPolicy.New)
                     .build();
             PushSubscribeOptions options = PushSubscribeOptions.builder()
                     .configuration(consumerConfig)
                     .build();
             JetStreamSubscription sub = jsConnection.subscribe(subject, dispatcher,
-                msg -> onMessage(msg, messageClass, messageConsumer), false, options);
+                msg -> onMessage(subject, msg, messageClass, messageConsumer), false, options);
```

Comment:
Type Message has a method called `getSubject()`. Maybe we can use that instead of passing a new arg

Reply:
would you mind running this and double checking that the NATS message.getSubject() matches the subject string passed as an argument to our subscribe method?

Reply:
My initial hesitation was that `.getSubject()` might be some internal subject string that NATS uses under the hood

Reply:
yes of course!

Reply:
We could just leave it the way it is since its a working version. Should i still test it out?